linter.py

```diff
@@ -177,6 +204,8 @@ def find_errors(self, output):
             filename = entry.get('filePath', None)
             if filename == '<text>':
                 filename = 'stdin'
+            elif filename.split(os.path.sep)[-1].split('.')[0] == BUFFER_FILE_STEM:
```

Comment:
As we already import `os` it is more isiomatic to use `os.path.basename` here. E.g. 
```suggestion
            elif filename and os.path.basename(filename).startswith(BUFFER_FILE_STEM + '.'):
```

Reply:
There are various other ways to do this but filename can be `None` here so an `elif filename and ...` is necessary in all cases!